syft/cataloger/cataloger.go

```diff
@@ -33,16 +33,30 @@ type Cataloger interface {
 	// TODO: we should consider refactoring to return a set of io.Readers instead of the full contents themselves (allow for optional buffering).
 }
 
-// All returns a slice of all locally defined catalogers (defined in child packages).
-func All() []Cataloger {
+// ImageCatalogers returns a slice of locally implemented catalogers that are fit for detecting installations of packages.
+func ImageCatalogers() []Cataloger {
 	return []Cataloger{
-		dpkg.New(),
-		bundler.New(),
-		python.New(),
-		rpmdb.New(),
-		java.New(),
-		apkdb.New(),
-		golang.New(),
-		javascript.New(),
+		ruby.NewGemSpecCataloger(),
+		python.NewPythonCataloger(),         // TODO: split and replace me
+		javascript.NewJavascriptCataloger(), // TODO: split and replace me
+		deb.NewDpkgdbCataloger(),
+		rpmdb.NewRpmdbCataloger(),
+		java.NewJavaCataloger(),
+		apkdb.NewApkdbCataloger(),
+		golang.NewGoModCataloger(),
+	}
+}
+
+// DirectoryCatalogers returns a slice of locally implemented catalogers that are fit for detecting packages from index files (and select installations)
+func DirectoryCatalogers() []Cataloger {
+	return []Cataloger{
+		ruby.NewGemFileLockCataloger(),
+		python.NewPythonCataloger(),         // TODO: split and replace me
+		javascript.NewJavascriptCataloger(), // TODO: split and replace me
+		deb.NewDpkgdbCataloger(),
```

Comment:
If for `apk` the package name is `apkdb`, shouldn't this be `dpkgdb` instead of `deb`? I guess this is implying there are multiple different ways of inspecting `deb` packages and we are only using `dpkgdb` here. 
The new consistency makes sense for the constructors, but I'm not sure what the idea is with the packages

Reply:
good point --there is an inconsistency here that only was partially adressed. There was an attempt to make the package names more generic to the category, while allowing for more specific catalogers. How about:

- `deb.NewDpkgdbCataloger()`
- `apk.NewApkdbCataloger()`
- `rpm.NewrpmdbCataloger()`

This falls more in line with the `python`, `ruby`, `javascript` package names, which encapsulate an ecosystem, and at the same time doesn't prevent us from having more catalogers within the package (e.g. both `rpm.NewRpmCataloger()` and `rpm.NewRpmdbCataloger()`).